pages/apiPages/moderator_page.ts

```diff
@@ -0,0 +1,304 @@
+import { APIRequestContext, expect, request } from "@playwright/test"
+import { Headers } from "../../utils/headers"
+import { faker } from '@faker-js/faker';
+import { apiDataSet } from "../../utils/dataSet";
+import { ModetarorPayloads } from "./moderator_payloads";
+
+
+export class ApiModeratorPage {
+    apiContext: APIRequestContext
+
+    constructor(apiContext: APIRequestContext) {
+        this.apiContext = apiContext
+    }
+
+    async createNewModerator(url: string, adminToken: string, email: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = ModetarorPayloads.createNewModerator(email)
+
+        const headers = Headers.userHeader(adminToken)
+
+        const apiRequest = await apiContext.post(`${url}:3000/admin/profile/create`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const response = await apiRequest.json()
+        const id = response._id
+        const rolesGroup = response.rolesGroup
+        expect(rolesGroup).toEqual('admin')
+        console.log(`Admin with: ${id}, ${email} and "password" is created`)
+        return { id, email }
+    }
+
+    async moderatorLogin(url: string, guestUserToken: string, email: string, deviceId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = ModetarorPayloads.moderatorLogin(guestUserToken, email, deviceId )
+        const headers = Headers.guestHeader()
+
+        const apiRequest = await apiContext.post(`${url}:3000/admin/login`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const response = await apiRequest.json()
+        const id = response.profile._id
+        const rolesGroup = response.profile.rolesGroup
+        const newAdminToken = response.token
+        expect(rolesGroup).toEqual('admin')
+        console.log(`Admin with: ${id}, ${email} and "password" is Logged In`)
+        return { id, email, newAdminToken }
+    }
+
+
+
+    async getAdminReferralEarnings(url: string, userToken: string, userId: string) {
+        const apiContext = await request.newContext({ignoreHTTPSErrors: true})
+        const data = {
+            "period": "day",
+            "userId": `${userId}`
+        }
+        const headers = Headers.userHeader(userToken)
+
+        const apiRequest = await apiContext.post(`${url}:3000/admin/agent/referalEarnings`, {data, headers: headers})
+        expect(apiRequest.ok()).toBeTruthy()
+        const response = await apiRequest.json()
+        const returnedUserId = response.user._id
+        expect(userId).toEqual(returnedUserId)
+        console.log(`Correct ${returnedUserId} is displayed`)
+    }
+
+    async setAdminProfileStatus(url: string, adminToken: string, referralUserId: string, currentStatus: string, newStatus: string) {
```

Comment:
@YulianH newStatus should be a predefine value

Reply:
done